### ISSUE TITLE
How to finetune with my own dataset?

### ISSUE THREAD

**Issue by thiner** [Vision-CAIR/MiniGPT-4#159]
Thanks for your great work and kindly open source to the public community! 
I am wondering how to finetune(or maybe other correct terms) this model with my own dataset? I read the [training](https://github.com/Vision-CAIR/MiniGPT-4#training) description, but I still have questions:

1. Should I repeat the two stages if I want to tune the model? Can I jump into stage 2 directly with the [checkpoint ](https://drive.google.com/file/d/1u9FRRBB3VovP1HxCAlpD9Lw4t4P6-Yq8/view?usp=share_link)file your provided and my dataset?
2. Need I append my own dataset to [dataset folder](https://drive.google.com/file/d/1nJXhoEcy3KTExr17I7BXqY5Y9Lx_-n-9/view?usp=share_link) provided at stage 2, or just replace it?

Thanks in advance!

**Comment by eeeuu** [comment]
dataset/README_2_STAGE.md

**Comment by TsuTikgiau** [comment]
Hello! Our stage 2 focuses on teaching the model to talk in a natural way. If your own dataset can also serve for this purpose, you can ignore our stage2 dataset. If your dataset focuses on improving the model's ability to see something, I guess you can insert an additional training stage using your own dataset between stage 1 and stage 2.

**Comment by thiner** [comment]
> Hello! Our stage 2 focuses on teaching the model to talk in a natural way. If your own dataset can also serve for this purpose, you can ignore our stage2 dataset. If your dataset focuses on improving the model's ability to see something, I guess you can insert an additional training stage using your own dataset between stage 1 and stage 2.

Thanks a lot for your guidance!
Regarding the middle step between stage 1 and stage 2, I guess I should follow the stage 2 manual to setup, am I right?

**Comment by TsuTikgiau** [comment]
@thiner Our original stage1 and stage2 use the same training script train.py. The difference is in the config file. In your case, you can create a new training config file based on stage2's config file, change the dataset in the config, modify the hyperparameter, etc. Also, when you want to add a new dataset, you might want to check how the datasets in this repository are built. Check the folder minigpt4/datasets to get some ideas. Thanks!

**Comment by thiner** [comment]
@TsuTikgiau Thanks a lot, really helpful!

**Comment by thiner** [comment]
> Could you successfully fine-tune the 7B model? I met problem about padding.

Yes, I ran the fine tuning stage(stage 2) sucessfully, but the effect is not good till not, I am still trying to make it work. 
Regarding "padding issue", I believe there are many relevant issues raised by other users, you may search for solution.

**Comment by Zenobia7** [comment]
@thiner Do you see a way to evaluate the model? I've been poking around the code for a long time without looking at the code and introduction to model evaluation and validation data sets

**Comment by cnxupupup** [comment]
> same question

**Comment by SLKAlgs** [comment]
@TsuTikgiau hello！May I know where is the evaluation dataset "cc_sbu_align" for stage2? I encountered this prompt when running stage2: "No validation splits found." How is the generated text validated? Additionally, I would like to fine-tune stage2 with my own dataset. Do I only need to replace "MiniGPT-4/train_configs/minigpt4_stage2_finetune.yaml" with my own dataset in the "minigpt4_stage2_finetune.yaml" file to use it?

**Comment by SLKAlgs** [comment]
@thiner hello！May I know where is the evaluation dataset "cc_sbu_align" for stage2? I encountered this prompt when running stage2: "No validation splits found." How is the generated text validated? Additionally, I would like to fine-tune stage2 with my own dataset. Do I only need to replace "MiniGPT-4/train_configs/minigpt4_stage2_finetune.yaml" with my own dataset in the "minigpt4_stage2_finetune.yaml" file to use it?

**Comment by zhangwengyu999** [comment]
Same question. How to evalute the the finetuning process? I set `evaluate: True` in the  `minigpt4_stage2_finetune.yaml`, but it says `No validation splits found.`. Where and how should I put my validation splits? Thank you.

**Comment by Davidwhw** [comment]
> Same question. How to evalute the the finetuning process? I set `evaluate: True` in the `minigpt4_stage2_finetune.yaml`, but it says `No validation splits found.`. Where and how should I put my validation splits? Thank you.

The verification step between training multiple epochs does not appear to be implemented in minigpt's code (such as lavis `valid_step`). 
Have you found a solution to your problem?

**Comment by lixiaoqingnnz** [comment]
@TsuTikgiau Hi, I would like to know if the ckpt "MiniGPT-v2 (after stage-2)" and "MiniGPT-v2 (after stage-3)" had trained on the datasets that you mentioned in the following link? 
https://github.com/Vision-CAIR/MiniGPT-4/blob/main/dataset/README_MINIGPTv2_FINETUNE.md

Thanks so much
